src/darwin/frida-helper-process.vala

```diff
@@ -380,7 +380,7 @@ namespace Frida {
 					service.stop ();
 
 				if (died_unexpectedly)
-					throw new Error.PROCESS_NOT_FOUND ("Peer process died unexpectedly");
+					throw new Error.PROCESS_NOT_FOUND ("Peer process (%d) died unexpectedly: %s", pending_pid, e.message);
```

Comment:
Line exceeds 140 cols -- needs to be wrapped, with scope added (since the statement will span multiple lines).